odoo/addons/base/models/ir_attachment.py

```diff
@@ -500,12 +501,25 @@ def unlink(self):
 
     @api.model_create_multi
     def create(self, vals_list):
+        location = self._storage()
         record_tuple_set = set()
         for values in vals_list:
             # remove computed field depending of datas
             for field in ('file_size', 'checksum'):
                 values.pop(field, False)
             values = self._check_contents(values)
+            # inline the inverse of datas
+            if 'datas' in values:
+                data = values.pop('datas')
+                bin_data = base64.b64decode(data) if data else b''
+                values['file_size'] = len(bin_data)
+                values['checksum'] = self._compute_checksum(bin_data)
+                values['index_content'] = self._index(bin_data, values['mimetype'])
+                values['store_fname'] = False
+                values['db_datas'] = data
+                if data and location != 'db':
+                    values['store_fname'] = self._file_write(data, values['checksum'])
+                    values['db_datas'] = False
```

Comment:
factor this in a method, also used by inverse_datas

Reply:
Done